pink/barriers/barrier.py

```diff
@@ -208,6 +223,9 @@ def compute_qp_inequality(
         :math:`\dot{q}` is the joint velocity vector,
         and :math:`\alpha_j` are extended class K functions.
 
+        Note:
+            Jacobian and barrier values are cached to avoid recomputation.
```

Comment:
I'm not against the idea of caching (a.k.a. memoization), but there are some reasons why it has not been done in Pink so far:

* We assume the robot is moving most of the time.
* The IK should run fast enough for control without memoization (because in the worst case, the robot just moved and we run the full computation anyway).
* It increases code complexity:
    * a bit more code for maintainers to handle 
    * more parameters to future-proof use cases from other users (see `np.allclose` below)
    * more logic for future developers to keep in mind (e.g. someone developing a new barrier may will have to know this to understand why their custom barrier/Jacobian functions are not called all the time)

Have you measured an improvement from caching in your tests so far?

Reply:
The main reason why we've decided to implement caching is that we are generally computing both cost and constraint for the same iteration, which results in computing Jacobian and bias twice per iteration. 

This was particularly severe, as the controller struggled to keep up even at 50Hz. So, this was the main purpose of caching: not to skip updates between pink's iterations, but to skip updates during one pink iteration.

Reply:
Probably this is not the best solution possible, but I failed to come up with something more descent without changing the logic of `solve_ik`.

Any suggestions on this matter are welcome!